### PR TITLE
ISPN-14896 Validate argument size with arity

### PR DESCRIPTION
**Pull request by karesti**
https://issues.redhat.com/browse/ISPN-14896

### CODE REVIEW

**Comment by karesti** [comment]
The change discovered a bug in PING arity value

File: server/resp/src/main/java/org/infinispan/server/resp/Resp3AuthHandler.java [hunk 1]
```diff
@@ -21,14 +21,19 @@ public Resp3AuthHandler(RespServer server) {
    }
 
    @Override
-   protected CompletionStage<RespRequestHandler> actualHandleRequest(ChannelHandlerContext ctx, RespCommand type, List<byte[]> arguments) {
-      if (type instanceof AuthResp3Command) {
-         AuthResp3Command type1 = (AuthResp3Command) type;
-         return type1.perform(this, ctx, arguments);
+   protected CompletionStage<RespRequestHandler> actualHandleRequest(ChannelHandlerContext ctx, RespCommand command, List<byte[]> arguments) {
+      if (command instanceof AuthResp3Command) {
+         if (command.hasValidNumberOfArguments(arguments)) {
```
Comment (jabolina) [inline server/resp/src/main/java/org/infinispan/server/resp/Resp3AuthHandler.java:26]: We can avoid the `hasValidNumberOfArguments` in each of the handlers and have it just on the `RespRequestHandler#handleRequest` before invoking the `actualHandleRequest`.

**Review by jabolina — COMMENTED** [review]
Just one suggestion.

File: server/resp/src/main/java/org/infinispan/server/resp/Resp3AuthHandler.java (hunk 1 above)
Comment (karesti) [reply server/resp/src/main/java/org/infinispan/server/resp/Resp3AuthHandler.java:26]: I had a doubt on this one to be honest !

**Review by jabolina — APPROVED** [review]

**Comment by jabolina** [comment]
Integrated into main, thanks @karesti!

**Pull request merged**